### PR TITLE
Pull and Load E2E Images

### PR DESCRIPTION
**Pull request by roopeshvs**
#### What type of PR is this?
/kind flake

#### What this PR does / why we need it:
Fixes flaky tests that were timing out as the images used in e2e were not being pulled in time.

#### Which issue(s) this PR fixes:
This PR addresses [Issue - Pre-pull docker images used for testing](https://github.com/kubernetes-sigs/kueue/issues/1108). The issue contains more information on the flaky tests.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: hack/e2e-test.sh [hunk 1]
```diff
@@ -55,6 +55,11 @@ function startup {
 function kind_load {
     if [ $CREATE_KIND_CLUSTER == 'true' ]
     then
+        export E2E_IMAGES=$(grep '.Image(' ./test/e2e/e2e_test.go | cut -d '"' -f2 | sort | uniq)
+        while read -r E2E_IMAGES; do
+            docker pull $E2E_IMAGES
+            $KIND load docker-image $E2E_IMAGES --name $KIND_CLUSTER_NAME
+        done <<< "$E2E_IMAGES"
```
Comment (trasc) [inline hack/e2e-test.sh:62]: For me, parsing the .go file is a bit too much, we currently have only one additional image. Also, I don't think that going forward there is any guarantee that  `e2e_test.go` will remain the only file of interest or that the potentially new images will also be provided as string literals and not const or var.

Comment (roopeshvs) [reply hack/e2e-test.sh:62]: @trasc I understand that parsing the go file to pick the image names is not ideal but there doesn't seem to be any other source for these image names.
Should we add a variable in `e2e-test.sh` in where people can manually add the image names that need to be pre-pulled?
Do you have any recommendation?

Comment (alculquicondor) [reply hack/e2e-test.sh:62]: We probably will only use one image that gives us good control over the behavior (how long it runs, which exit code it returns). Putting it in a variable should be enough.

**Comment by tenzen-y** [comment]
fixes: #1108

**Pull request merged**